sprinter/tests/test_environment.py

```diff
@@ -295,4 +308,12 @@ def test_feature_run_remove_failure(self):
 namespace = testsprinter
 my_custom_value = bar
 non_custom_value = baz
+inputs = my_custom_value==fee
```

Comment:
The new logic is better in that values are only carried over if the input exists in the target. No more orphaned values!